rascaline/src/math/kvectors.rs

```diff
@@ -0,0 +1,133 @@
+// Generate the k-vectors (also called reciprocal or Fourier vectors)
+// needed for the k-space implementation of LODE and SOAP.
+// More specifically, these are all points of a (reciprocal space)
+// lattice that lie within a ball of a specified cutoff radius.
+use crate::{Matrix3, Vector3D};
+
+/// Kvectors
+#[derive(Debug, Clone)]
+pub struct KVector {
+    /// 3 component k-vector
+    pub vector: Vector3D,
+    /// length of the k-vector
+    pub norm: f64,
+}
+
+pub fn compute_kvectors(cell: &Matrix3, kcutoff: f64) -> Vec<KVector> {
+    // Generate k-vectors up to a certain cutoff (in reciprocal space units) 
+    // for a given cell.
+    let reciprocal_cell = 2.0 * std::f64::consts::PI * cell.transposed().inverse();
```

Comment:
You should add a sanity check that the cell is defined, something like 

```suggestion
/// Generate k-vectors up to a certain cutoff (in reciprocal space units) 
/// for a given cell.
pub fn compute_kvectors(cell: &Matrix3, kcutoff: f64) -> Vec<KVector> {
    debug_assert_ne!(cell, Matrix3::zero(), "invalid cell matrix");

    let reciprocal_cell = 2.0 * std::f64::consts::PI * cell.transposed().inverse();
```

Or take a `UnitCell` as input and check the shape here!

Reply:
I like the approach with the matrix, because this is really general. Thanks for the sanity checks!